Changes to Typescript Quickstart (#3582)
kubeshop/tracetest: docs/docs/tools-and-integrations/typescript.mdx

**Docker**: Have [Docker](https://docs.docker.com/get-docker/) and [Docker Compose](https://docs.docker.com/compose/install/) installed on your machine.

> ## Run this Quckstart example

Reviewer: ```suggestion
## Run This Quckstart Example
```

The example below is provided as part of the Tracetest project. You can download and run the example by following these steps:

> Clone the Tracetest project and go to the Typescript Quickstart:

Reviewer 2: ```suggestion
Clone the Tracetest project and go to the TypeScript Quickstart:
```

```bash
git clone https://github.com/kubeshop/tracetest
cd tracetest/examples/quick-start-typescript
```

> Follow these instruction to run the included demo app and typescript example:

Reviewer 2: ```suggestion
Follow these instructions to run the included demo app and TypeScript example:
```

1. Copy the `.env.template` file to `.env`.
2. Log into the [Tracetest app](https://app.tracetest.io/).

> 3. This example is configured to use of the OpenTelemetry Collector. Ensure the environment you will be utlizing to run this example is also configured to use the OpenTelemetry Tracing Backend by clicking on Settings, Tracing Backend, OpenTelemetry, Save.

Reviewer: ```suggestion
3. This example is configured to use the OpenTelemetry Collector. Ensure the environment you will be utilizing to run this example is also configured to use the OpenTelemetry Tracing Backend by clicking on Settings, Tracing Backend, OpenTelemetry, Save.
```

4. Fill out the [token](https://docs.tracetest.io/concepts/environment-tokens) and [agent API key](https://docs.tracetest.io/concepts/agent) details by editing your .env file. You can find these values in the Settings area for your environment.
5. Run `docker compose up -d`.
6. Look for the `tracetest-client` service in Docker and click on it to view the logs. It will show the results from the trace-based tests that are triggered from the index.ts Typescript file.

> 7. Follow the links in the log to to view the test runs programmatically created by your Typescript program.

Reviewer 2: ```suggestion
7. Follow the links in the log to to view the test runs programmatically created by your TypeScript test script.
```

> The sections below breakdown in detail what the example you just ran did and how it works.

Reviewer 2: ```suggestion
Follow along with the sections below for an in detail breakdown of what the example you just ran did and how it works.
```

## Project Structure

> The quick start Typescript project is built with Docker Compose.

Reviewer 2: ```suggestion
The quick start TypeScript project is built with Docker Compose.
```

> The [Pokeshop Demo App](/live-examples/pokeshop/overview) is a complete example of a distributed application using different backend and front-end services. We will be launching it and running tests against it as part of this example.

Reviewer 2: ```suggestion
The [Pokeshop Demo App](/live-examples/pokeshop/overview) is a complete example of a distributed application using different back-end and front-end services. We will be launching it and running tests against it as part of this example.
```

The `docker-compose.yaml` file in the root directory of the quick start runs the Pokeshop Demo app, the OpenTelemetry Collector setup, and the [Tracetest Agent](/concepts/agent).

> The Typescript Quickstart has two primary files: a Typescript file `definitions.ts` that defines two Tracetest tests, and a Typescript file 'index.ts' that imports these test definitions and uses the "@tracetest/client" NPM package to run them multiple times. 

Reviewer 2: ```suggestion
The TypeScript quick start has two primary files: a Typescript file `definitions.ts` that defines two Tracetest tests, and a Typescript file 'index.ts' that imports these test definitions and uses the "@tracetest/client" NPM package to run them multiple times. 
```

> We will show you how to install the NPM package and use these two Typescript programs to programatically run Tracetest tests.

Reviewer 2: ```suggestion
We will show you how to install the NPM package and use these two TypeScript files to programmatically run Tracetest tests.
```

## Installing the `@tracetest/client` NPM Package

> The first step when using the Typescript NPM package is to install the `@tracetest/client` NPM Package. It is as easy as running the following command:

Reviewer 2: ```suggestion
The first step when using the TypeScript NPM package is to install the `@tracetest/client` NPM Package. It is as easy as running the following command:
```

main();
```

## Running the Full Example

To start the full setup, run the following command:

```bash
docker compose up -d
```

## Finding the Results

> The output from the Typescript script should be visible in the log for the `tracetest-client` service via Docker Desktop. This log will show links to Tracetest for each of the test runs invoked by index.ts. Click a link to launch Tracetest and view the test result.

Reviewer 2: ```suggestion
The output from the TypeScript script should be visible in the log for the `tracetest-client` service in Docker Compose. This log will show links to Tracetest for each of the test runs invoked by `index.ts`. Click a link to launch Tracetest and view the test result.
```
